exercises/worker-versioning/README.md

```diff
@@ -0,0 +1,160 @@
+# Exercise 2: Version the Change with Worker Versioning
+
+During this exercise, you will
+
+- Define Worker Build ID Version Sets and enable Versioning on your Worker
+- Make a change to your Workflow, and redeploy an updated Worker
+- Ensure that your Workflows have switched to using the new code path
+- Decommission an old Worker
+
+If you are running this exercise locally with a Temporal Cluster that
+you started via the `temporal server start-dev` command, you must
+restart it using some additional parameters that will enable the
+Worker Versioning feature:
+
+```shell
+temporal server start-dev \
+   --dynamic-config-value frontend.workerVersioningDataAPIs=true \
+   --dynamic-config-value frontend.workerVersioningWorkflowAPIs=true \
+   --dynamic-config-value worker.buildIdScavengerEnabled=true
+```
+
+The above step is not necessary if you are using GitPod to run this
+exercise.
+
+Make your changes to the code in the `practice` subdirectory (look for
+`TODO` comments that will guide you to where you should make changes to
+the code). If you need a hint or want to verify your changes, look at
+the complete version in the `solution` subdirectory.
+
+## Part A: Assign a Build ID to Your Worker and Task Queue
+
+Before enabling Worker Versioning, run this Workflow without making any further
+modifications. Unlike our other exercises, the `practice` subdirectory contains
+fully working example code.
+
+1. Navigate to `exercises/patch-versioning/practice` by running
+   `cd exercises/patch-versioning/practice`
+   1. If you're in the GitPod environment you can run the shortcut command `ex1`
+1. Launch the Worker in one terminal by running the command `mvn exec:java -Dexec.mainClass="pizzaworkflow.PizzaWorker"`
+   1. If you're in the GitPod environment you can run the shortcut command `ex1w`
+1. Start the Workflow in another terminal by running the command `mvn exec:java -Dexec.mainClass="pizzaworkflow.Starter"`
+   1. If you're in the GitPod environment you can run the shortcut command `ex1st`
+1. This creates an environment where an unverioned Worker is already running.
+1. Edit the `PizzaWorker.java` file to add a Build ID to your Worker and opt in
+   to Worker Versioning.
+
+   1. Add the following code to specify `WorkerOptions` for you worker, specifying
+      the `BuildID` as `v2` and enabling versioning by setting `UseBuildIdForVersioning` to `true`
+      ```java
+      WorkerOptions workerOptions = WorkerOptions.newBuilder()
+         .setBuildId("v2")
+         .setUseBuildIdForVersioning(true)
+         .build();
+      ```
+
+1. Edit the `Stater.java` to inform the Task Queue about your Worker's Build ID before starting your Workflow.
+   ```java
+   client.updateWorkerBuildIdCompatability(Constants.TASK_QUEUE_NAME,
```

Comment:
Might want to not you don't need to do this every time you run the starter, only needs to happen once.

Reply:
Going to chat with the team about this. Can you confirm that doing this multiple times is not harmful, only unnecessary?

Reply:
So for now we're going to add the following statement 
```
**Note:** This code would usually only need to be run once. In a production system you would not run this as part of your client, but more likely as part of your build system on initial deployment, either via the SDK or the Command Line Tool
```
Underneath it. Next year we're going to go back and reevaluate our use of the word `Starter` to describe things and then maybe pull this out.

Reply:
Makes sense, the risk is if you deploy another group of workers with a new version (say v3) you may have two groups of workers both trying to call `updateWorkerBuildIdCompatability`

Reply:
That's good to know. I'll take note of this